### PR TITLE
feat:Add advertisement banners to the forum

### PR DESCRIPTION
**Pull request by oranges-eating**
An ad slot has been added to the right of the site, and the administrator can modify it in the background

### CODE REVIEW

**Comment by hsluoyz** [comment]
@oranges-eating plz fix:

image

**Comment by why-mark** [comment]
> @ oranges-eating plz修复：
> 
> 图像

already solved

**Comment by oranges-eating** [comment]
> @oranges-eating plz fix:
> 
> image

already solved

File: conf/app.conf [hunk 1]
```diff
@@ -3,7 +3,7 @@ httpport = 7000
 runmode = dev
 SessionOn = true
 copyrequestbody = true
-dataSourceName = root:123@tcp(localhost:3306)/
+dataSourceName = root:12345678@tcp(localhost:3308)/
```
Comment (kocoler) [inline conf/app.conf]: Why change this field?

File: controllers/poster.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	_ "encoding/json"
+	"fmt"
+	"github.com/casbin/casnode/object"
+	_ "github.com/casbin/casnode/object"
+	_ "github.com/casbin/casnode/util"
+	"io/ioutil"
+	_ "io/ioutil"
+	"os"
```
Comment (kocoler) [inline controllers/poster.go]: Fmt your package import.

Comment (kocoler) [inline controllers/poster.go:25]: Fmt your package import.

Comment (kocoler) [inline controllers/poster.go:25]: Fmt your package import.

File: controllers/poster.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	var resp Response
+	if !object.CheckModIdentity(c.GetSessionUser()) {
+		resp = Response{Status: "fail", Msg: "Unauthorized."}
+		c.Data["json"] = resp
+		c.ServeJSON()
+		return
+	}
+    poster := string(c.Ctx.Input.RequestBody)
+    w,err :=os.OpenFile("./dictionary/poster.json",os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0644)
+    if err != nil{
```
Comment (kocoler) [inline controllers/poster.go]: Using gofmt to format your code.

Comment (kocoler) [inline controllers/poster.go:38]: Using gofmt to format your code.

File: controllers/poster.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		panic(err)
+	}
+	defer r.Close()
+	poster, err := ioutil.ReadAll(r)
+	if err != nil{
+		panic(err)
+	}
+	c.Data["json"] = string(poster)
+	c.ServeJSON()
+}
```
Comment (kocoler) [inline controllers/poster.go]: There should be a new line after each file.

File: controllers/poster.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		return
+	}
+    poster := string(c.Ctx.Input.RequestBody)
+    w,err :=os.OpenFile("./dictionary/poster.json",os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0644)
+    if err != nil{
+    	panic(err)
+	}
+	w.WriteString(poster)
+    defer w.Close()
+	fmt.Println(poster)
```
Comment (kocoler) [inline controllers/poster.go]: Remove useless log.

File: controllers/poster.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	c.Data["json"] = Response{Status: "ok", Msg: "success"}
+	c.ServeJSON()
+}
+
+func (c *APIController) ReadPoster() {
+	r,err := os.Open("./dictionary/poster.json")
```
Comment (kocoler) [inline controllers/poster.go]: I think you should read the advertisement from DB, not a local file.

File: web/src/admin/AdminPoster.js [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        }
+
+        this.state.url = `/admin/poster`;
+    }
+
+    componentDidMount() {
+    }
+
+    updateposter() {
+        PosterBackend.updateposter_info(this.state.memberId, this.state.form).then(
```
Comment (kocoler) [inline web/src/admin/AdminPoster.js]: Why memberId?

File: service/mail.go [hunk 8]
```diff
@@ -64,6 +65,7 @@ func SendRegistrationMail(email, validateCode string) error {
 	mail := gomail.NewMessage()
 
 	name := beego.AppConfig.String("appname")
+	fmt.Println(validateCode)
```
Comment (kocoler) [inline service/mail.go]: `mail.go` has no related to your pr.

File: controllers/poster.go (hunk 6 above)
Comment (kocoler) [reply controllers/poster.go]: Same as the storage part above.

**Review by kocoler — CHANGES_REQUESTED** [review]
Remember to format your code before creating a pr.

File: web/src/admin/AdminPoster.js (hunk 7 above)
Comment (why-mark) [reply web/src/admin/AdminPoster.js:64]: > 为什么是memberId？
Use member id to determine whether it can be changed

Comment (oranges-eating) [reply web/src/admin/AdminPoster.js]: > 为什么是memberId？

Use member id to determine whether it can be changed

**Pull request closed without merging**